src/reconciler/spec/reconciler.rs

```diff
@@ -45,7 +47,11 @@ pub trait Reconciler<#[verifier(maybe_negative)] K: ResourceView>: Sized {
     // Lib should contain method process, it should implement a trait (which should be the spec version of ExternalLibrary).
     // It must be a generic currently. This will cause another round of super annoying refactoring. So currently we just
     // add another method to this trait.
-    open spec fn external_process(input: Self::I) -> Option<Self::O>;
+    // This method consumes the input (which should be computed by reconcile_core) and the current state of the external
+    // library and produces the response and the next state of the library.
+    // Use optional state here because: (1) it's easy to initialize since we don't have to require a default or init method,
+    // (2) some libraries don't need a state to hold information, thus, optional state makes sense. 
+    open spec fn external_process(input: Self::I, state: Option<Self::S>) -> (Option<Self::O>, Option<Self::S>);
```

Comment:
I am actually not sure about using option here. If the API is stateless, then S can just be `()`, and defining a init method for S as part of `reconciler` should be fine. We can live with the option thing if you think using the option is really better.

Reply:
Yes, I am not sure about this either. When I was thinking about adding an init method, I just didn't like the Reconciler trait contains 2 out of 6 methods to describe the external library. But using Option also has some downsides.